Cookbook/Proposals/ManagingAssets.md

```diff
@@ -0,0 +1,54 @@
+
+To make sure we won't have duplicated icons in the project and all icons will be accessible in all places where we implement the user interface, all of us should agree on uniform rules how and where we should add new assets and how we should manage them.
+
+
+## Proposed solution
+
+The proposed solution contains a set of rules which should be followed during manipulation of our assets:
+1. All new icons and images we are going to use should be added to `BabylonDependencies.framework` in `Assets.xcassets` catalog 
+2. `struct DesignLibrary.Tokens` should have another property for icons `public let icons: Icons`
+3. Icons should be namespaced and put into `struct` with common relation/meaning (e.g. `TabBarIcons`, `Iconography` etc.). If there is no existing `struct` we should create one. Then we have to add a new case to `enum ImageIdentifier` which is embedded in given `struct`.
+4. To access e.g. iconography's close icon we could use subscripts `designLibrary.tokens.icons.iconography[.close]`
+5. If we are updating an icon we should check if any other place uses it. If so and that other place shouldn't be updated we should create a new icon with an updated design.
+6. If we are removing some code which is using some icons we should check if that code was the last place which was using given icon, if so icon should be deleted from `Assets.xcassets`.
+7. To support the overriding of standard icons in white label apps we should update `ImageCatalogueAware.image(for imageIdentifier:in bundle: compatibleWith traitCollection:) -> UIImage` to firstly access image from `Bundle.main` and then fallback to `BabylonDependencies` if image was not overridden. We can also add support to specify from which location we would like to access given image.
+
+## Impact on existing codebase
+
+Unfortunately, in our codebase icons and images are located in different places. If we will agree on above set of rules, they should be applied for newly created assets. We should also make an effort to eliminated technical debt and migrate all existing icons into `BabylonDependencies`.
+Keeping every icon and image in one location potentially shouldn’t increase the size of the target application on the condition that the app has all our frameworks linked.
+## Alternatives considered
+
+1. We could try to systematize the way we include assets in specific feature frameworks but it can cause problems described in the motivation section.
+
+2. Instead of accessing icons or images by its identifier `designLibrary.tokens.icons.iconography[.close]` we could create computed properties for each icon:
+```
+public struct Iconography: ImageCatalogueAware
+	...
+	public var close: UIImage {
+	    return Iconography.image(for: "close")
+	}
+}
+```
+then use `designLibrary.tokens.icons.iconography.close`.
+That solution will require to write more code per each icon. There is no extra value here because autocomplition should also work with `ImageIdentifier`'s as subscript argument.
```

Comment:
we won't need to write those properties in Swift 5.1 though as it supports key paths in `@dynamicMemberLookup`. Favourite feature of @andersio